src/components/ControlledForm/index.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { ControlledInputProps } from '../Input';
+
+export interface Field {
+  name: string;
+  value?: any;
+  isValid?: boolean;
+}
+
+type Props = {
+  initialValues?: object;
+  onSubmit: (values) => void;
+  render: (formState: {
+    isValid: boolean;
+    state: { [name: string]: Field };
+    getInputControls: <T = string>(name: string) => ControlledInputProps<T>;
+  }) => React.ReactNode;
+};
+
+const ControlledForm = (props: Props) => {
+  const [fields, setFields] = useState<{ [key: string]: Field }>({});
+
+  const isValid = !Object.values(fields).filter(field => !!field.isValid).length;
```

Comment:
```suggestion
  const isValid = Object.values(fields).every(field => !!field.isValid);
```

Reply:
hey @victor0402 here, if there is any field isValid undefined, the field should be valid. In this case, I need to check before, because the props onValidationChange is optional.

Reply:
`const isValid = Object.values(fields).every(
    field => field.isValid === undefined || field.isValid
  );` WDYT?

Reply:
I see, looks good to me